### ISSUE TITLE
Training DetectNet with GTX1070+ubuntu 14.04+Intel i7+8G Mem

### ISSUE THREAD

**Issue by David-Jin** [NVIDIA/DIGITS#966]
My machine is Intel i7(3.2GHz) + GTX1070(8GB Mem) + Ubuntu14.04 + 8G System Memory+Cuda8.0+CuDnn V5+ pre-trained berkley model

I want to run the demo of object detect. I train the DetectNet according to the parameter of tutorial.
But it says it will take almost 20 hours.  The lr is set to 0.0001.  After training 8 hours, the process is 43%.
when it trained epoch 12, the accuracy is still 0.2.  But in the tutorial, when on epoch 5, the accuracy rise to 80%. The total process of tutorial is 7 Hours.

It that normal?  something wrong? Generally, How long should the object detect trainning should take ?

I want to post my desktop's snapshot, but git issues seems do not support uploading pictures.

**Comment by David-Jin** [comment]
I post my picture of desktop snapshot on google groups:

https://groups.google.com/d/msg/digits-users/iAukkGudG68/VDzOa0M5BQAJ

I know there are someone else also come across this problem. Can anyone give  advice?

**Comment by lasandcris** [comment]
I think that sounds about right due to the size of the images in the KITTI dataset. For me it took almost 24 hours to run the example using the GTX1080. But then I used my own dataset with images (400x600), 4000 for training and 1500 for validation, single class object detection. This took almost 2 hours. 

Hope this helps.

**Comment by lukeyeager** [comment]
> Generally, How long should the object detect trainning should take ?

tl;dr: It's going to take a while. You're probably not doing anything wrong.

Depending on several factors, I've seen DetectNet trainings on my machine go as fast as 2.5 hours and as slow as 4 days. Several of those trainings never converged to any useful result. As @lasandcris pointed out, some of the relevant factors are the size and number of images you're using. Here are all the things I can think of that you could do to speedup training:

**Data**

* Use smaller images
* Use fewer images

**Network**

* Use a larger stride for the grid

**Solver**

* Train for fewer epochs

**Software**

* Use the latest version of NVcaffe (currently `0.15.13`)
* Use the latest version of cuDNN (currently `5.1.5-1+cuda8.0` or `5.1.3-1+cuda7.5`)

**Hardware**

* Use multiple GPUs
* Get a newer, faster GPU (Pascal > Maxwell > Kepler)
* Get a GPU with more memory so you can use a bigger batch size and/or use faster convolutional kernels from cuDNN
* Put your data on a RAID for faster IO
* Get a CPU that's faster and/or has more cores

Of course, some of those suggestions are counter-productive if you're also having trouble getting the network to train. Hopefully most of this is obvious - these are mostly just generic suggestions for machine learning.
